frontend/app/src/scenes/Home/scenes/UpsertCategory/index.tsx

```diff
@@ -50,7 +57,18 @@ class UpsertCategory extends React.Component<
       )
       .then((result: AxiosResponse<Category>) => {
         if (result.status == 200) {
-          this.setState({ isLoading: false, category: result.data });
+          // Gets the translations for the fetched category and setting them to the state using a map
+          const categoryTranslations = new Map();
+          result.data.translations.forEach((translation) => {
+            categoryTranslations.set(translation.language, {
+              name: translation.name,
+            });
+          });
+          this.setState({
+            translations: categoryTranslations,
+            isLoading: false,
+            category: result.data,
+          });
         } else {
           throw new Error();
```

Comment:
you have to set isLoading false and show an error notification to the user

Reply:
oh god. sorry I didn't see it. throwing an error is sufficient because it is being caught by the catch block.